### PR TITLE
[RLlib] Docs do-over (new API stack): Prep. RLModule; introduce `Default[algo]RLModule` classes (rename from `[algo]RLModule`).

### PR DESCRIPTION
**Pull request by sven1977**
Introduce `Default[algo]RLModule` classes (rename from `[algo]RLModule`).
* This indicates a deeper move towards a world, where users don't have to subclass a algo-specific RLModule superclass anymore, but rather "only" have to implement APIs xyz in order to fulfil an algo's requirements.
* For example, in order to learn with PPO, a user now can provide any(!) arbitrary RLModule, given that it implements the `ValueFunctionAPI`. The user does NOT have to subclass a PPO-specific superclass anymore freeing them from hidden/confusing/underlying implementation details, such as
  * In which property is the encoder?
  * Do I have to worry about that property if I want to bring in my own encoder or not use an encoder altogether?

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/appo/default_appo_rl_module.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from ray.rllib.core.rl_module.apis import TargetNetworkAPI
+from ray.rllib.utils.typing import NetworkType
+
+from ray.rllib.utils.annotations import (
+    override,
+    OverrideToImplementCustomLogic_CallToSuperRecommended,
+)
+
+
+class DefaultAPPORLModule(DefaultPPORLModule, TargetNetworkAPI, abc.ABC):
```
Comment (simonsays1980) [inline rllib/algorithms/appo/default_appo_rl_module.py:18]: I like this default naming. Do we apply it to all algorithms now? That would be great!

File: rllib/algorithms/bc/bc.py [hunk 2]
```diff
@@ -70,10 +70,12 @@ def __init__(self, algo_class=None):
...
     def get_default_rl_module_spec(self) -> RLModuleSpecType:
         if self.framework_str == "torch":
-            from ray.rllib.algorithms.bc.torch.bc_torch_rl_module import BCTorchRLModule
+            from ray.rllib.algorithms.bc.torch.default_bc_torch_rl_module import (
+                DefaultBCTorchRLModule,
+            )
 
             return RLModuleSpec(
-                module_class=BCTorchRLModule,
+                module_class=DefaultBCTorchRLModule,
```
Comment (simonsays1980) [inline rllib/algorithms/bc/bc.py:78]: Awesome!

File: rllib/algorithms/dqn/dqn_rainbow_rl_module.py [hunk 3]
```diff
@@ -119,11 +119,6 @@ def forward_target(self, batch: Dict[str, Any]) -> Dict[str, Any]:
             ),
         )
 
-    # TODO (simon): DQN Rainbow does not support RNNs, yet.
```
Comment (simonsays1980) [inline rllib/algorithms/dqn/dqn_rainbow_rl_module.py:122]: This comes with the sequence sampling PR. Have to finish it.

File: rllib/algorithms/ppo/default_ppo_rl_module.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        # If this is an `inference_only` Module, we'll have to pass this information
+        # to the encoder config as well.
+        if self.inference_only and self.framework == "torch":
+            self.catalog.actor_critic_encoder_config.inference_only = True
+
+        # Build models from catalog.
+        self.encoder = self.catalog.build_actor_critic_encoder(framework=self.framework)
+        self.pi = self.catalog.build_pi_head(framework=self.framework)
+        self.vf = self.catalog.build_vf_head(framework=self.framework)
+        # __sphinx_doc_end__
```
Comment (simonsays1980) [inline rllib/algorithms/ppo/default_ppo_rl_module.py:47]: Do we still need this sphinx logic here? All other modules don't have it.

File: rllib/offline/offline_data.py [hunk 5]
```diff
@@ -69,8 +69,9 @@ def __init__(self, config: AlgorithmConfig):
             self.filesystem_object = self.filesystem
         elif self.filesystem is not None:
             raise ValueError(
-                f"Unknown filesystem: {self.filesystem}. Filesystems can be "
-                "'gcs' for GCS, 's3' for S3, or 'abs'"
+                f"Unknown `config.input_filesystem` {self.filesystem}! Filesystems "
```
Comment (simonsays1980) [inline rllib/offline/offline_data.py:72]: Nice.

**Review by simonsays1980 — APPROVED** [review]
LGTM. Some nice changes that clarify a bit more that users can start from our default RL modules for customization.

File: rllib/algorithms/appo/default_appo_rl_module.py (hunk 1 above)
Comment (sven1977) [reply rllib/algorithms/appo/default_appo_rl_module.py:18]: Yeah, exactly, that's the idea! Only DQN and SAC missing, I think ...

File: rllib/algorithms/ppo/default_ppo_rl_module.py (hunk 4 above)
Comment (sven1977) [reply rllib/algorithms/ppo/default_ppo_rl_module.py:47]: So far yes. A follow up PR will redo the rst files and the example codes. WIP ...

File: rllib/algorithms/appo/default_appo_rl_module.py (hunk 1 above)
Comment (sven1977) [reply rllib/algorithms/appo/default_appo_rl_module.py:18]: DQN and the rest will be done in the [next PR](https://github.com/ray-project/ray/pull/49368) (already in review ...)

**Pull request merged**